json/src/value.rs

```diff
@@ -1333,3 +1512,139 @@ impl<T: ?Sized> ToJson for T
         to_value(&self)
     }
 }
+
+/// Just like `std::borrow::Borrow<Value>` but borrowing may fail.
+pub trait TryBorrow<'a>: 'a + ::std::marker::Sized {
+    /// Returns reference to internal data if type matches.
+    fn try_borrow(value: &'a Value) -> Result<Self, TypeError>;
```

Comment:
The reason is to mimic `Borrow` and `BorrowMut`. I'm not even sure whether it would be possible to have them together in one trait, since one is defined for `&'a T` and the other for `&'a mut T`...

However, I'm open to put them together, if it's possible and advantageous.